stores/auctionsStore.js

```diff
@@ -4,14 +4,17 @@ import { AUCTION_DATA_FETCHER } from '../constants';
 
 const initialPageState = { pageStart: 0, pageEnd: 10, pageStep: 10 };
 
-const transformEvents = async (auctions, service) => {
+const transformEvents = async (auctions, service, type, ilk) => {
   const groupedEvents = _.groupBy(auctions, auction => auction.auctionId);
   let auctionsData = {};
-  console.log('fetched events')
+  console.log('fetched events');
   await Promise.all(
     Object.keys(groupedEvents).map(async id => {
       try {
-        const { end, tic } = await service.getFlopDuration(id);
+        const { end, tic } =
+          type === 'flip'
+            ? await service.getFlipDuration(id, ilk)
+            : await service.getFlopDuration(id);
```

Comment:
Does this maybe justify a `getDuration` convenience method within the service that internally calls `getFlipDuration` or `getFlopDuration`?

Reply:
yep, much cleaner